add readme example to encourage contributors to highlight `with_options` (#112)
PrefectHQ/prefect-collection-template: {{cookiecutter.collection_name}}/README.md

    hello_{{ cookiecutter.collection_slug }},
)

> # Use `with_options` to customize settings on a task or flow defined in {{ cookiecutter.collection_slug }}:

Reviewer: Maybe we can mention `any` since it's not just for collection tasks/flows 
```suggestion
# Use `with_options` to customize settings on any existing task or flow:
```

custom_goodbye_{{ cookiecutter.collection_slug }} = goodbye_{{ cookiecutter.collection_slug }}.with_options(
    name="My custom task name",

>     max_retries=2,

Reviewer: ```suggestion
    retries=2,
```

    retry_delay_seconds=10,
)

@flow
def example_flow():
    hello_{{ cookiecutter.collection_slug }}
    custom_goodbye_{{ cookiecutter.collection_slug }}

example_flow()
```

> For more tips on how to use tasks and flows in a Collection, check out [Using Collections](https://orion-docs.prefect.io/collections/usage/)!

Reviewer: Can you also make a PR in Prefect as a new heading to mention all this?

example_flow()
```

To adjust settings on a pre-configured task or flow defined in {{ cookiecutter.collection_slug }}, use `with_options`:

```python
from prefect import flow

> from {{ cookiecutter.collection_slug }}.tasks import task_defined_in_{{ cookiecutter.collection_slug }}

Reviewer: Since the template contains the hello and goodbye task, we should use that. Also, maybe we can consolidate into the previous example for `goodbye` and add a comment saying this overrides.

Reviewer: Lastly, I think we can document this in the main docs since the example from here will not propagate to all our collections (since we have custom examples).
https://orion-docs.prefect.io/collections/usage/
